src/main/java/com/monumental/services/SessionFactoryService.java

```diff
@@ -1,28 +1,52 @@
 package com.monumental.services;
 
-import com.monumental.models.*;
+import com.monumental.models.Monument;
+import com.monumental.triggers.MonumentTrigger;
 import org.hibernate.SessionFactory;
 import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
 import org.hibernate.cfg.Configuration;
+import org.hibernate.event.service.spi.EventListenerRegistry;
+import org.hibernate.event.spi.EventType;
+import org.hibernate.internal.SessionFactoryImpl;
 import org.hibernate.service.ServiceRegistry;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import javax.annotation.PostConstruct;
+import java.util.Arrays;
+import java.util.List;
+
 @Service
 public class SessionFactoryService {
 
+    @Autowired
+    public MonumentTrigger monumentTrigger;
+
+    /** IMPORTANT: Update these lists when creating triggers or models
+     *  Triggers must be autowired, and the string must match the property's name
+     */
+    private final List<String> triggers = Arrays.asList(
+        "monumentTrigger"
+    );
+    private final List<Class> models = Arrays.asList(
+        Monument.class
+    );
+
     private SessionFactory factory;
     private ServiceRegistry registry;
 
+    // These must correspond with the events covered by ModelTrigger
+    private static EventType[] eventTypes = new EventType[]{
+        EventType.PRE_INSERT, EventType.POST_INSERT, EventType.PRE_UPDATE,
+        EventType.POST_UPDATE, EventType.PRE_DELETE, EventType.POST_DELETE
+    };
+
     public SessionFactoryService() {
 
         try {
             Configuration conf = new Configuration()
                     .configure();
-            Class[] classes = new Class[]{
```

Comment:
Since we removed this Array, aren't we now missing `Model.class`? It's not added to `this.models`

Reply:
good catch